src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -62,10 +68,25 @@ public boolean equals(Object obj) {
   @Override
   public int hashCode() {
     int hash = 7;
-    hash = 31 * hash + (int) salience;
```

Comment:
... and here you can use `Double.doubleToLongBits()`. That's what `Double.compare()` uses to compare the numbers. Since in both `equals` and `hashCode` you will be comparing the floating-point representation of the numbers, you'll achieve the consistency Jerome mentioned above.

![image](https://user-images.githubusercontent.com/673690/93444449-8236d280-f8c8-11ea-9128-34bf93da2b7c.png)